R/sample_data.R

```diff
@@ -69,9 +69,9 @@ read_sample_data <- function(p, results_dir, tx2gene = NULL) {
   kallisto <- p[["kallisto"]]
 
   # check which quant input is provided
-  if (!isnull(salmon)) {
+  if (is.null(salmon)) {
     counts <- salmon_counts(salmon, tx2gene = tx2gene)
-  } else if ((!isnull(kallisto))) {
+  } else if ((is.null(kallisto))) {
```

Comment:
Why did the `!` disappear?

Reply:
I think the logic needs a bit more work here. What are the different options you're expecting? One can provide:

- a kallisto file
- a salmon file
- a salmon and a kallisto file
- neither

Reply:
For the last two options - would it suffice if "Null" is returned for these cases - that is:

```
else if (!is.null(kallisto) && !is.null(salmon)) {
    return(NULL)
  } else {
    return(NULL)
  }
```

Or we should generate specific error messages in the main rmd?

Reply:
Do you think the `dragen_wts_dir` should be removed and instead rnasum should just work directly with file paths? That would simplify this a bit in terms of logic. You'd just be dealing with a list `p` which may or may not have elements `salmon` and `kallisto`.

Reply:
> Do you think the dragen_wts_dir should be removed and instead rnasum should just work directly with file paths?

That'd then also require refactoring at infrastructure level as we use that as an input to CWL workflow?
Not negating - but perhaps something we should add as a to-do for the next release.

Reply:
For the last two options - how about assigning null to `counts` and then an assert that statement with an error message in rmd before `RNAsum::combineDatasets` call? So it doesn't proceed with further processing?

Reply:
If that makes sense please go ahead, sorry if I'm being a bit picky.
If I were to do this properly, I'd create a new function (e.g. `check_params`) that takes as input the `params` list, checks each element's validity, and outputs a new list with e.g. a `counts` element that gets used in the Rmd. But happy for you to implement the kallisto support however you wish, just try to make sure the basic cases are supported and that the logic works. Also I'd try stop the Rmd execution as early as possible upon invalid param specification.

Reply:
Spent sometime testing with the following variable conditions to make sure rmd execution is stopped earlier if the quant inputs are incorrect (ref: last two commits).
- kallisto file
- salmon file
- salmon and kallisto file
- neither